tests/jballerina-unit-test/src/test/java/org/ballerinalang/nativeimpl/jvm/servicetests/ServiceValue.java

```diff
@@ -166,7 +167,7 @@ private static String generateMethodName(BString methodName, ArrayValue path) {
         for (int i = 0; i < path.size(); i++) {
             funcName.append("$").append(path.getBString(i).getValue());
         }
-        return funcName.toString();
+        return IdentifierUtils.encodeFunctionIdentifier(funcName.toString());
```

Comment:
Instead of this, shouldn't the `getName` methods of the method types also return the decoded name?

Reply:
I think the type information is also encoded to reflect their counterpart in generated JVM code.